vite_ruby/lib/vite_ruby/runner.rb

```diff
@@ -31,7 +31,11 @@ def command_for(args)
       args = args.clone
       cmd.push('node', '--inspect-brk') if args.delete('--inspect')
       cmd.push('node', '--trace-deprecation') if args.delete('--trace_deprecation')
-      cmd.push(vite_executable)
+      if ViteRuby.commands.yarn_berry?
+        cmd.push('yarn', 'vite')
+      else
+        cmd.push(vite_executable)
+      end
```

Comment:
Let's move this check to `vite_executable`, modifying it to return an array:

```ruby
cmd.push(*vite_executable)
```

We would still want a user to be able to configure `vite_bin_path` as needed:

```ruby
  # Internal: Resolves to an executable for Vite.
  def vite_executable
    bin_path = config.vite_bin_path
    return [bin_path] if File.exist?(bin_path)
 
    if config.root.join('yarn.lock').exist?
      ['yarn', 'vite']
    else
      ["#{ `npm bin`.chomp }/vite]
    end
  end
```

That way we can more easily extend this for other package managers that don't use a `node_modules` structure.

Reply:
Thank you for comment! 👍